kernel/kernel-api/src/test/scala/io/delta/kernel/internal/types/DataTypeJsonSerDeSuite.scala

```diff
@@ -245,6 +274,249 @@ object DataTypeJsonSerDeSuite {
+        structFieldJson("c14", "\"variant\"", false)
+      )),
+      new StructType()
+        .add("c1", BinaryType.BINARY, true)
+        .add("c2", BooleanType.BOOLEAN, false)
+        .add("c3", ByteType.BYTE, false)
+        .add("c4", DateType.DATE, true)
+        .add("c5", DecimalType.USER_DEFAULT, false)
+        .add("c6", DoubleType.DOUBLE, false)
+        .add("c7", FloatType.FLOAT, false)
+        .add("c8", IntegerType.INTEGER, true)
+        .add("c9", LongType.LONG, true)
+        .add("c10", ShortType.SHORT, true)
+        .add("c11", StringType.STRING, true)
+        .add("c12", TimestampNTZType.TIMESTAMP_NTZ, false)
+        .add("c13", TimestampType.TIMESTAMP, false)
+        .add("c14", VariantType.VARIANT, false)
+    ),
+    (
+      structTypeJson(Seq(
+        structFieldJson("a1", "\"string\"", true),
+        structFieldJson("a2", structTypeJson(Seq(
+          structFieldJson("b1", mapTypeJson(
+            arrayTypeJson(
+              arrayTypeJson(
+                "\"string\"", true), true),
+            structTypeJson(Seq(
+              structFieldJson("c1", "\"string\"", false),
+              structFieldJson("c2", "\"string\"", true)
+            )), true), true),
+          structFieldJson("b2", "\"long\"", true))), true),
+        structFieldJson("a3",
+          arrayTypeJson(
+            mapTypeJson(
+              "\"string\"",
+              structTypeJson(Seq(
+                structFieldJson("b1", "\"date\"", false)
+              )), false), false), true)
+      )),
+      new StructType()
+        .add("a1", StringType.STRING, true)
+        .add("a2", new StructType()
+          .add("b1", new MapType(
+            new ArrayType(
+              new ArrayType(StringType.STRING, true), true),
+            new StructType()
+              .add("c1", StringType.STRING, false)
+              .add("c2", StringType.STRING, true), true))
+          .add("b2", LongType.LONG), true)
+        .add("a3", new ArrayType(
+          new MapType(
+            StringType.STRING,
+            new StructType()
+              .add("b1", DateType.DATE, false), false), false), true)
+    )
+  )
+
+  val SAMPLE_JSON_TO_TYPES_WITH_COLLATION = Seq(
+    (
+      structTypeJson(Seq(
+        structFieldJson("a1", "\"string\"", true,
+          metadataJson = Some(s"""{"$COLLATIONS_METADATA_KEY" : {"a1" : "ICU.UNICODE"}}""")),
+        structFieldJson("a2", "\"integer\"", false),
+        structFieldJson("a3", "\"string\"", false,
+          metadataJson = Some(s"""{"$COLLATIONS_METADATA_KEY" : {"a3" : "SPARK.UTF8_LCASE"}}""")),
+        structFieldJson("a4", "\"string\"", true))),
+      new StructType()
+        .add("a1", new StringType("ICU.UNICODE"), true)
+        .add("a2", IntegerType.INTEGER, false)
+        .add("a3", new StringType("SPARK.UTF8_LCASE"), false)
+        .add("a4", StringType.STRING, true)
+    ),
+    (
+      structTypeJson(Seq(
+        structFieldJson("a1", structTypeJson(Seq(
```

Comment:
nit: add a comment just above the `structFieldJson` on what test this field is covering. Easy to see the tests and understand.